symforce/slam/imu_preintegration/manifold_symbolic.py

```diff
@@ -0,0 +1,346 @@
+        DR_D_gyro_bias (sf.M33): Derivative of DR w.r.t. gyro_bias
+        Dv_D_gyro_bias (sf.M33): Derivative of Dv w.r.t. gyro_bias
+        Dv_D_accel_bias (sf.M33): Derivative of Dv w.r.t. accel_bias
+        Dp_D_gyro_bias (sf.M33): Derivative of Dp w.r.t. gyro_bias
+        Dp_D_accel_bias (sf.M33): Derivative of Dp w.r.t. accel_bias
+        gyro_bias (sf.V3): Initial guess for the gyroscope measurement bias
+        accel_bias (sf.V3): Initial guess for the accelerometer measurement bias
+        gyro_cov (sf.M33): The covariance of the gyroscope measurement
+        accel_cov (sf.M33): The covariance of the accelerometer measurement
+        gyro (sf.V3): The gyroscope measurement
+        accel (sf.V3): The accelerometer measurement
+        dt (T.Scalar): The time between the new measurements and the last
+        epsilon (T.Scalar): epsilon for numerical stability
+
+    Returns:
+        T.Tuple[sf.Rot3, sf.V3, sf.V3, sf.Matrix, sf.Matrix, sf.Matrix, sf.Matrix, sf.Matrix, sf.Matrix]:
+            new_DR,
+            new_Dv,
+            new_Dp,
+            new_covariance,
+            new_DR_D_gyro_bias,
+            new_Dv_D_gyro_bias,
+            new_Dv_D_accel_bias,
+            new_Dp_D_gyro_bias,
+            new_Dp_D_accel_bias
+    """
+    # Correct for IMU bias
+    corrected_accel = accel - accel_bias
+    corrected_gyro = gyro - gyro_bias
+
+    # Integrate the state
+    new_DR, new_Dv, new_Dp = integrate_state(
+        DR,
+        Dv,
+        Dp,
+        gyro=corrected_gyro,
+        accel=corrected_accel,
+        dt=dt,
+        epsilon=epsilon,
+    )
+
+    # NOTE(Brad): Both ways of calculating the derivatives of new_state should be the same.
+    if False:  # pylint: disable=using-constant-test
+        # Definitely correct, but not very amenable to CSE
+        def new_state_D(wrt_variables: T.List[T.Any]) -> sf.Matrix:
+            return sf.Matrix.block_matrix(
+                [
+                    tangent_jacobians(new_DR, wrt_variables),
+                    tangent_jacobians(new_Dv, wrt_variables),
+                    tangent_jacobians(new_Dp, wrt_variables),
+                ]
+            )
+
+        new_state_D_state = new_state_D([DR, Dv, Dp])
+        new_state_D_gyro_bias = new_state_D([gyro_bias])
+        new_state_D_accel_bias = new_state_D([accel_bias])
+    else:
+        # Handwritten derivatives, reduces op count by ~16%
+        (
+            new_state_D_state,
+            new_state_D_gyro,
+            new_state_D_accel,
+        ) = handwritten_state_D_old_state_gyro_accel(
+            DR, corrected_gyro, corrected_accel, dt, epsilon
+        )
+        new_state_D_gyro_bias = -new_state_D_gyro
+        new_state_D_accel_bias = -new_state_D_accel
+
+    new_covariance = new_state_D_state * covariance * new_state_D_state.T
+    new_covariance += new_state_D_gyro_bias * (gyro_cov / dt) * new_state_D_gyro_bias.T
```

Comment:
This should technically be `new_covariance += new_state_D_gyro * (gyro_cov / dt) * new_state_D_gyro.T`?

Reply:
+1

Reply:
Technically, yes. However, since `new_state_D_gyro` is just `-new_state_D_gyro_bias`, it doesn't actually matter which we use here. I just did it this way because I figured might as well reuse the one value if I already have it.

However, as you guys just demonstrated, the code is more readable if `new_state_D_gyro` is used (as that way you won't need to know the relation between `new_state_D_gyro` and `new_state_D_gyro_bias` for it to make sense); so I've changed it to be as you wrote.

Incidentally, this actually reduced the op count marginally for the handwritten case.